src/VirtoCommerce.XCatalog.Data/Queries/GetBrandQueryHandler.cs

```diff
@@ -70,4 +77,30 @@ public async Task<BrandAggregate> Handle(GetBrandQuery request, CancellationToke
 
         return brand;
     }
+
+    private async Task<Category> GetBrandCategory(GetBrandQuery request, Catalog brandsCatalog)
+    {
+        var result = default(Category);
+
+        if (!request.Id.IsNullOrEmpty())
+        {
+            result = await _categoryService.GetNoCloneAsync(request.Id);
+        }
+        else if (!request.Name.IsNullOrEmpty())
+        {
+            // If Id is not specified, then the brand will be resolved by Name.
+            var categorySearchCriteria = AbstractTypeFactory<CategorySearchCriteria>.TryCreateInstance();
+            categorySearchCriteria.CatalogId = brandsCatalog.Id;
+            categorySearchCriteria.Keyword = request.Name;
```

Comment:
There may be more matching categories than criteria.Take. If so, the search results may not include the category with the requested name.